src/lib.rs

```diff
@@ -117,10 +117,53 @@ use macos as platform;
 /// this sparingly.
 ///
 /// [c_rehash]: https://www.openssl.org/docs/manmaster/man1/c_rehash.html
-pub fn load_native_certs() -> Result<Vec<CertificateDer<'static>>, Error> {
-    match CertPaths::from_env().load()? {
-        Some(certs) => Ok(certs),
-        None => platform::load_native_certs(),
+pub fn load_native_certs() -> CertificateResult {
+    match CertPaths::from_env().load() {
+        out if !out.certs.is_empty() => out,
+        _ => platform::load_native_certs(),
+    }
+}
+
+#[derive(Debug, Default)]
```

Comment:
how about `#[non_exhaustive]`, so we could add items later in a semver-compatible release?